app.js

```diff
@@ -70,17 +72,30 @@ app.post('/sessions', async function(req, res, next) {
     if (process.env['LOG_SINK_URL'])
       request.post({ url: process.env['LOG_SINK_URL'], body: tokenSet, json: true });
 
-    const { groupUri, groupId } = await selectBestuurseenheidByNumber(claims);
+    let { groupUri, groupId } = await selectGroupByNumber(claims);
+
+    const isEconomischeActor = claims.vo_doelgroepcode == "EA"
 
     if (!groupUri || !groupId) {
-      console.log(`User is not allowed to login. No bestuurseenheid found for roles ${JSON.stringify(claims[roleClaim])}`);
-      saveLog(
-        logsGraph,
-        `http://data.lblod.info/class-names/no-bestuurseenheid-for-role`,
-        `User is not allowed to login. No bestuurseenheid found for roles ${JSON.stringify(claims[roleClaim])}`,
-        sessionUri,
-        claims[groupIdClaim]);
-      return res.header('mu-auth-allowed-groups', 'CLEAR').status(403).end();
+      if(isEconomischeActor) {
+        await createEconomischeActorByClaims(claims);
+        ({ groupUri, groupId } = await selectGroupByNumber(claims));  
+
+
+        if (!groupUri || !groupId) {
+          console.log(`Error: GroupUri and GroupID are still empty even after creating them! Claims = ${JSON.stringify(claims)}`);
+          return res.header('mu-auth-allowed-groups', 'CLEAR').status(500).end();
+        }
```

Comment:
Seems to be a case that cannot occur unless there is a (consistent) bug in the implementation. Doesn't need to be double checked here imo as that problem should have bubbled up during development/testing.